### PR TITLE
Added Documentation

### PR DESCRIPTION
**Pull request by grokys**
- Added "getting started" and "contributing" guides
- Added a "welcome" message linking to docs from Team Explorer

### CODE REVIEW

**Comment by jcansdale** [comment]
It looks like the documentation here is out of date:
https://services.github.com/on-demand/windows/visual-studio

It suggests assessing the GitHub Extension by selecting 'View > Other Windows > GitHub' from the main menu (which is currently just the PR list). I should maybe mention something about the icons and repo list on the 'Team Explorer' pane.

I hear @beardofedu might have something to do with these pages? 😄

**Comment by paladique** [comment]
Just listing out the additional things to add to the docs, feel free to add more if I missed anything:

- [ ] View on GitHub (open/copy/blame)
    - [ ] Open
    - [ ] Copy
    - [ ] Blame
- [ ] Removing repos from VS (so they don't show up in TE, under GitHub)
- [ ] Tool bar
    - [ ] Refresh
    - [ ] Pull Requests
    - [ ] GitHub button
- [ ] Settings

File: docs/contributing/index.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Use GitHub for Visual Studio to manage your projects and work with Pull Requests.
+
+### Table of Contents
+
+- Adding and cloning repositories
+  - [Publishing an existing project to GitHub](publishing-an-existing-project-to-github.md)
+  - [Creating an empty repository from Visual Studio](creating-an-empty-repository-from-visual-studio.md)
+  - [Cloning a repository to Visual Studio](cloning-a-repository-to-visual-studio.md)
+- Working with Pull Requests
+  - [View the Pull Requests for a repository](view-the-pull-requests-for-a-repository.md)
```
Comment (bernars) [inline docs/contributing/index.md]: > View the Pull Requests for a repository

Your other procedurals are titled with gerunds (which is consistent with the GitHub Help) - can this be "Viewing pull requests for a repository" to make it parallel?

To be consistent with GitHub Help, I recommend not capitalizing "issues" or "pull requests" in general (https://manual-of-style.githubapp.com/capitalization/#product-features):

> *A note on feature capitalization: Use your best judgment when using title case to refer to product features. For example, use sentence case if capitalizing a feature in the middle of the sentence looks jarring or inconsistent.*

File: docs/contributing/index.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  - [Sending a Pull Request](sending-a-pull-request.md)
+  - [Review a Pull Request in Visual Studio](review-a-pull-request-in-visual-studio.md)
```
Comment (bernars) [inline docs/contributing/index.md]: > Review a Pull Request in Visual Studio

To make this parallel with other titles, "Review" should be "Reviewing".

File: docs/contributing/index.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  - [Making changes to a Pull Request](making-changes-to-a-pull-request.md)
+- Operations provided by Visual Studio
+  - [Committing](https://www.visualstudio.com/en-us/docs/git/tutorial/commits)
+  - [Pushing](https://www.visualstudio.com/en-us/docs/git/tutorial/pushing)
+  - [Fetching and pulling](https://www.visualstudio.com/en-us/docs/git/tutorial/pulling)
+  - [Work with branches](https://www.visualstudio.com/en-us/docs/git/tutorial/branches)
+  - [View history](https://www.visualstudio.com/en-us/docs/git/tutorial/history)
```
Comment (bernars) [inline docs/contributing/index.md]: > View history

To make this parallel with other titles, this should be "Viewing history".

File: docs/contributing/making-changes-to-a-pull-request.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Making changes to a Pull Request
+
+When a Pull Request branch is [checked out](review-a-pull-request-in-visual-studio.md) you can commit changes to it and push and pull like any other branch. If the Pull Request branch is located in a fork and was checked out from the GitHub Pull Request details pane, then a remote to that fork will be created automatically and the branch set to track the fork branch.
```
Comment (bernars) [inline docs/contributing/making-changes-to-a-pull-request.md]: > a Pull Request branch

I've not often heard this called a "pull request branch"; it seems more accurate to call it a topic branch.

File: docs/contributing/making-changes-to-a-pull-request.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+## Pulling Changes
```
Comment (bernars) [inline docs/contributing/making-changes-to-a-pull-request.md]: > Pulling Changes

I think the concept of pulling is easier to understand if you expand on this, maybe:

> Pulling changes to your local clone

File: docs/contributing/making-changes-to-a-pull-request.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+If a Pull Request is checked out and the author adds new commits to the branch, then the option will be given to pull the changes locally. This works both for Pull Requests from the same repository and from a fork.
+
+TODO: screenshot
+
+## Pushing Changes
+
+If you make commits locally to a Pull Request branch then you can push the changes to the remote branch. You can also do this from `git` itself or from the Visual Studio Team Explorer **Sync** view.
```
Comment (bernars) [inline docs/contributing/making-changes-to-a-pull-request.md]: > `git` itself

The command is called `git`, but the product name is Git (capitalized, not monospaced).

File: docs/contributing/making-changes-to-a-pull-request.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+> Note: for this to work with Pull Requests that come from forks, then you must be a maintainer on the repository and the Pull Request submitter must have checked [Allow edits from maintainers](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) when submitting the Pull Request.
+
+If there are remote changes to the pull request branch then they must be pulled before you can push your changes.
```
Comment (bernars) [inline docs/contributing/making-changes-to-a-pull-request.md]: > If there are remote changes to the pull request branch then they must be pulled before you can push your changes.

What about:

"If there are commits on the branch on the remote repository that you don't have on your local clone, you must pull them to your local clone and resolve any conflicts before you can push your local commits back to the remote repository."

We have some articles for resolving merge conflicts (a daunting task to some) that you might consider linking to: https://help.github.com/articles/addressing-merge-conflicts/.

File: docs/contributing/publishing-an-existing-project-to-github.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# Publishing an existing project to GitHub
+
+1. Open a solution in Visual Studio.
+2. If solution is not already initialized as a git repository, select **Add to Source Control** from the **File** menu. TODO: screenshot
+3. Open **Team Explorer**. TODO: screenshot
+4. Click the **Sync** button in Team Explorer. TODO: screenshot
```
Comment (bernars) [inline docs/contributing/publishing-an-existing-project-to-github.md]: > Click the **Sync** button in Team Explorer.

How about:

"In Team Explorer, click **Sync**."

File: docs/contributing/publishing-an-existing-project-to-github.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+5. Click the **Publish to GitHub** button. TODO: screenshot
```
Comment (bernars) [inline docs/contributing/publishing-an-existing-project-to-github.md]: > Click the **Publish to GitHub** button.

We don't explicitly call out "button" or "link" or "menu", we just refer to the label in bold: https://manual-of-style.githubapp.com/typography/#ui-elements

File: docs/getting-started/authenticating-to-github.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+1. In Visual Studio, select **Team Explorer** from the **View** menu.
+![](images/view_team_explorer.png)
+2. In the Team Explorer pane, click the **Manage Connections** toolbar button.
+![](images/manage_connections.png)
+3. Click the **Connect** link in the GitHub section. If you are already connected to a GitHub instance and want to connect to another, this link will not be visible; instead click **Manage Connections** and then **Connect to GitHub**.
+![](images/connect_to_github.png)
+4. In the **Connect to GitHub dialog** choose **GitHub** or **GitHub Enterprise**, depending on which product you're using.
+5. Type your credentials, then click the **Login** button.
```
Comment (bernars) [inline docs/getting-started/authenticating-to-github.md]: > click the **Login** button.

This looks like it's referring to the action of signing in; can you change that UI text to match our terminology guidelines? https://manual-of-style.githubapp.com/terminology/#general-terminology

File: docs/getting-started/installing-github-for-visual-studio.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+1. Visit the [GitHub for Visual Studio](https://visualstudio.github.com/) site.
+2. Click the **Download GitHub Extension for Visual Studio** button.
+3. In your computer's **Downloads** folder, double-click **GitHub.VisualStudio.vsix**.
+TODO: Insert screenshot
+4. In the pop-up window, click **Install**.
+5. After the installation is completed, run Visual Studio.
+
+## Installing from the Visual Studio gallery
+
+If you're currently running Visual Studio 2015 or higher, you can install the extension from the Visual Studio gallery with the following steps:
```
Comment (bernars) [inline docs/getting-started/installing-github-for-visual-studio.md]: > If you're currently running Visual Studio 2015 or higher, you can install the extension from the Visual Studio gallery with the following steps:

You can omit "... with the following steps" here:

> If you're currently running Visual Studio 2015 or higher, you can install the extension from the Visual Studio gallery.

**Comment by paladique** [comment]
Thanks so much for the review @bernars! 

I've incorporated most of the [fixes suggested](https://github.com/github/VisualStudio/pull/935#pullrequestreview-29039844). Some of them require fixes in multiple pages, but the main takeaways are to:

#### Use "Sign in" instead of "Log in"

#### Orient the user, then tell them what to do when they get there 
For example,

> Next to the account you want to clone from, click Clone.

Instead of

>Click the Clone link next to the account

#### Be aware of where to capitalize
> To be consistent with GitHub Help, I recommend not capitalizing "issues" or "pull requests" in general.

#### You can omit button/link/menu vocabulary 
> We don't explicitly call out "button" or "link" or "menu", we just refer to the label in bold

**Comment by bernars** [comment]
Those are the main takeaways, @paladique! These are looking SO GOOD.

File: docs/contributing/cloning-a-repository-to-visual-studio.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
+# Cloning a repository to Visual Studio
+
+After you provide your GitHub or GitHub Enterprise credentials to GitHub for Visual Studio, the extension automatically detects the personal and organization repositories you have access to on your account.
+
+1. Open **Team Explorer**. TODO: screenshot
+2. Click the **Manage Connections** toolbar button. TODO: screenshot
+3. Click the **Clone** link next to the account you want to clone the repository from. TODO: screenshot
+4. In the list of repositories, click the repository you'd like to clone. TODO: screenshot
+5. Change the local path for the clone if necessary. TODO: screenshot
+6. Click the **Clone** button.
```
Comment (shana) [inline docs/contributing/cloning-a-repository-to-visual-studio.md]: It's right below it.

File: docs/contributing/making-changes-to-a-pull-request.md (hunk 4 above)
Comment (grokys) [inline docs/contributing/making-changes-to-a-pull-request.md]: What I was trying to refer from was the branch that the Pull Request was created from. Is that what "topic branch" means? I've never heard this terminology, but we can switch to that if it'd be better.

**Comment by paladique** [comment]
Should we include a FAQ page, like we do in on the [site](https://visualstudio.github.com/help.html)?

**Comment by bernars** [comment]
In general, I'm against FAQs. [18f describes why so beautifully that I always leave that to them](https://pages.18f.gov/content-guide/structure-the-content/#dont-use-faqs).

With that said, we did create [a VERY SMALL HANDFUL of FAQs for the GItHub Desktop site](https://help.github.com/desktop/faq/), and they were truly questions that Web Support was getting over and over again. You'll notice they follow a common pattern:

1. The answers are incredibly brief.
2. There's always a link to documentation with a more verbose answer.

Hope that's helpful!

File: docs/contributing/making-changes-to-a-pull-request.md (hunk 4 above)
Comment (paladique) [inline docs/contributing/making-changes-to-a-pull-request.md]: yeah I haven't either, but it's called "topic branch" in the [help docs](https://help.github.com/articles/creating-a-pull-request-from-a-fork/), so we should probably switch it

**Comment by shana** [comment]
This is looking SO GOOD! 🎉  ❤️ 
I'll get some screenshots for the "Installing" doc over the weekend and I think we're good to go on merging this.

**Comment by grokys** [comment]
@bernars @shana @paladique @jcansdale I think we're just about ready for v1.0 of docs. I've removed the WIP label, time for a final review!

**Comment by shana** [comment]
LGTM! One thing we should do is add descriptions to all `` tags, so that they are readable by screen readers and other accessibility software. I opened a separate issue #952 to track this, we can tackle that separately.

**Comment by shana** [comment]
image

**Comment by bernars** [comment]
CONGRATULATIONS! 👏 📚 💥

**Pull request merged**